Resources/Prototypes/Entities/Structures/Wallmounts/extinguisher_cabinet.yml

```diff
@@ -8,7 +8,12 @@
     - type: Sprite
       sprite: Structures/Wallmounts/extinguisher_cabinet.rsi
       netsync: false
-      state: extinguisher_closed
+      layers:
+      - state: frame
+      - state: extinguisher
+        map: ["enum.ItemCabinetVisuals.ContainsItem"]
+      - state: closed
+        map: ["enum.ItemCabinetVisualLayers.Door"]
```

Comment:
you're mixing layer keys, use ItemCabinetVisualLayers for all of these, make one for the item as well as the door